dockerfiles/Dockerfile.centos7

```diff
@@ -16,13 +16,15 @@ RUN yum -y update && \
 COPY . /src/
 WORKDIR /src/build
 
+ARG EXTRA_BUILD_ARGS
```

Comment:
Does this need rolling to all the package builds as well?
I would probably make a named variable for it rather than a generic one, e.g. `FLB_STATIC_CONF`, to make it more obvious to control.

Reply:
> Does this need rolling to all the package builds as well?

I'd say not - this is a build flag that requires a local build from source/Git. It can't be enabled in a packaged build (assuming I'm understanding packaged build correctly). The config option makes fluentbit embed the configuration files into the binary rather than looking for them from disk - so it's not something that can be distributed. 

The goal here was to have a test that ensure it still works (it's broken a few times in the past) - and the Centos7 build just seemed (arbitrarily) like an easy place to put it. Open to putting it somewhere else instead though :-) We've run into a few small compile bugs when `-DFLB_STATIC_CONF` has been enabled, so this would prevent those compile bugs into the future :-)

>  I would probably make a named variable for it rather than a generic one, e.g. FLB_STATIC_CONF, to make it more obvious to control.

Great call :-) Thanks for the suggestion :-)

Reply:
We can always add another container build to verify static builds directly and do this in parallel then as well.
This one was added to primarily verify the legacy target compilation worked so hijacking it for something else does not seem right or obvious.
I would add a new CI job just for this I think for one of the faster targets, you can just invoke the packaging build with FLB_ARGS set for it too I think to pass them in: https://github.com/fluent/fluent-bit/blob/8aee285464c30d1af03fdfbf1dcbdf784b5ace33/packaging/build.sh#L16

We could also make a static compilation test for all of the packaging builds via a matrix too in the future.

Reply:
Thanks for the suggestion!

I've probably not quite done what you're suggesting yet (I haven't quite grokked `call-build-linux-packages.yaml` yet) but did reuse build.sh :-) For now I've left it in `pr-compile-check.yaml` (although it does lightly feel like a rationally named workflow file? :thinking: )